src/words/environment/Environment.java

```diff
@@ -135,6 +135,24 @@ public WordsObject createObject(String objectName, String className, Position po
 		}
 	}
 	
+	public void removeObject(WordsObject object) {
+		object.clearReferers();
+		
+		for (HashMap<String, WordsObject> scope : objectsByName) {
+			if (scope.get(object.getObjectName()) != null) {
+				scope.remove(object.getObjectName());
+				break;
+			}
+		}
+		
+		for (WordsClass wordsClass : objectsByClass.keySet()) {
```

Comment:
Does this really need to be done as a loop?  objectsByClass is a HashMap from WordsClass to the HashSet.  And the WordsObject knows it's WordsClass.  So couldn't be just use the HashMap to lookup the appropriate HashSet, and then remove the WordsObject from that HashSet immediately?